lib/jwt/algos/ecdsa.rb

```diff
@@ -45,7 +46,7 @@ def curve_by_name(name)
 
         {
           algorithm: algorithm,
-          digest: algorithm.sub('ES', 'sha')
+          digest: algorithm.sub('ES', 'sha').sub('K', '')
```

Comment:
Would love to see the algorithm being explicit in the `NAMED_CURVES` constant instead of trying to preserve the numbers in the name. But that could be addressed in another PR.

Reply:
Totally agree. That stood out to me as well. I'd be happy to include that change in this PR as I do think the "K" substitution pushes the current approach too far.

Reply:
That sounds super, just go for it:)